text/0000-tracked-properties.md

```diff
@@ -0,0 +1,1083 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Tracked Properties
+
+## Summary
+
+Tracked properties introduce a simpler and more ergonomic system for tracking
+state change in Ember applications. By taking advantage of new JavaScript
+features, tracked properties allow Ember to reduce its API surface area while
+producing code that is both more intuitive and less error-prone.
+
+This simple example shows a `Person` class with three tracked properties:
+
+```js
+export default class Person {
+  @tracked firstName = 'Chad';
+  @tracked lastName = 'Hietala';
+
+  @tracked get fullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+}
+```
+
+### A Note on Decorator Stability
+
+Decorator support in Ember is crucial for the release of Ember Octane. Native
+classes are not usable without decorators, and they are a major part of the next
+edition. That said, decorators are still a stage 2 proposal in TC39, and merging
+support for them before stage 3 represents a risk.
+
+This RFC is being made now so that we can get early discussion out of the way,
+and agree on the general parameters of the `@tracked` decorator in Ember.
+Ideally, there will be no major changes before decorators move to stage 3, and
+we'll be able to merge this RFC shortly after it is announced.
+
+Alternatively, if decorators do not move forward in TC39 in the next few months,
+we may need to consider making an exception for tracked properties, or scoping
+down Ember Octane.
+
+## Terminology
+
+Because of the occasional overlap in terminology when discussing similar
+features, this document uses the following language consistently:
+
+- A **getter** is a JavaScript feature that executes a function to determine the
+  value of a property. The function is executed every time the property is
+  accessed.
+- A **computed property** is a property on an Ember object whose value is
+  produced by executing a function. That value is cached until one of computed
+  property's dependencies changes.
+- A **tracked property** refers to any property that has been instrumented with
+  `@tracked`, either a _tracked getter_ or a _tracked simple property_.
+- A **tracked getter** is a JavaScript getter that has been wrapped using the
+  tracked decorator
+- A **tracked simple property** is a regular, non-getter property that has been
+  wrapped using the tracked decorator.
+- The **legacy programming model** refers to the traditional Ember programming
+  model. It includes _legacy classes_, _computed properties_, _event listeners_,
+  _observers_, _property notifications_, and _curly components_, and more
+  generally refers to features that will eventually be removed from Ember.
```

Comment:
> generally refers to features that will eventually be removed from Ember

This implies that computed properties and event listeners will go away, that would be a separate RFC.